### PR TITLE
feat(compat) : implement `isSet/isMap`

### PR DESCRIPTION
**Pull request by chhw130**
i added `isSet`, `isMap` in compat.

### CODE REVIEW

File: benchmarks/performance/isMap.bench.ts [hunk 1]
```diff
@@ -1,9 +1,7 @@
 import { bench, describe } from 'vitest';
-import { isMap as isMapToolkit_ } from 'es-toolkit';
-import { isMap as isMapLodash_ } from 'lodash';
-
-const isMapToolkit = isMapToolkit_;
-const isMapLodash = isMapLodash_;
+import { isMap as isMapToolkit } from 'es-toolkit';
+import { isMap as isMapToolkitCompat } from 'es-toolkit/compat';
+import { isMap as isMapLodash } from 'lodash';
```
Comment (D-Sketon) [inline benchmarks/performance/isMap.bench.ts:4]:
```suggestion
import { isMap as isMapToolkit_ } from 'es-toolkit';
import { isMap as isMapToolkitCompat_ } from 'es-toolkit/compat'
import { isMap as isMapLodash_ } from 'lodash';

const isMapToolkit = isMapToolkit_;
const isMapToolkitCompat = isMapToolkitCompat_ ;
const isMapLodash = isMapLodash_;
```

File: benchmarks/performance/isSet.bench.ts [hunk 2]
```diff
@@ -1,9 +1,7 @@
 import { bench, describe } from 'vitest';
-import { isSet as isSetToolkit_ } from 'es-toolkit';
-import { isSet as isSetLodash_ } from 'lodash';
-
-const isSetToolkit = isSetToolkit_;
-const isSetLodash = isSetLodash_;
+import { isSet as isSetToolkit } from 'es-toolkit';
+import { isSet as isSetToolkitCompat } from 'es-toolkit/compat';
+import { isSet as isSetLodash } from 'lodash';
```
Comment (D-Sketon) [inline benchmarks/performance/isSet.bench.ts:4]:
```suggestion
import { isSet as isSetToolkit_ } from 'es-toolkit';
import { isSet as isSetToolkitCompat_ } from 'es-toolkit/compat';
import { isSet as isSetLodash_ } from 'lodash';

const isSetToolkit = isSetToolkit_;
const isSetToolkitCompat = isSetToolkitCompat_;
const isSetLodash = isSetLodash_;
```

File: benchmarks/performance/isMap.bench.ts (hunk 1 above)
Comment (chhw130) [reply benchmarks/performance/isMap.bench.ts:4]: @D-Sketon 
Thanks for the review!!

File: src/predicate/isMap.ts [hunk 3]
```diff
@@ -16,6 +16,6 @@
  * console.log(isMap(value3)); // false
  */
 
-export function isMap(value: unknown): value is Map<any, any> {
+export function isMap(value?: unknown): value is Map<any, any> {
```
Comment (raon0211) [inline src/predicate/isMap.ts:19]:
```suggestion
export function isMap(value: unknown): value is Map<any, any> {
```

I think our original library has to accept the argument here.

File: src/predicate/isSet.ts [hunk 4]
```diff
@@ -16,6 +16,6 @@
  * console.log(isSet(value3)); // false
  */
 
-export function isSet(value: unknown): value is Set<any> {
+export function isSet(value?: unknown): value is Set<any> {
```
Comment (raon0211) [inline src/predicate/isSet.ts:19]:
```suggestion
export function isSet(value: unknown): value is Set<any> {
```

**Review by raon0211 — APPROVED** [review]
Thanks!

File: src/predicate/isMap.ts (hunk 3 above)
Comment (chhw130) [reply src/predicate/isMap.ts]: I appreciate your confirmation!
Thank you 👍

**Pull request merged**